http-server-undertow/src/test/groovy/io/micronaut/servlet/undertow/EventsListener.java

```diff
@@ -8,6 +9,7 @@
 import io.micronaut.runtime.server.event.ServerStartupEvent;
 import jakarta.inject.Singleton;
 
+@Requires(property = "spec.name", value = "UndertowStartStopSpec")
```

Comment:
this is to avoid bean pollution introduced in this PR by @dstepanov https://github.com/micronaut-projects/micronaut-servlet/pull/352